plugins/agent_bridge_mcp_client.go

```diff
@@ -0,0 +1,434 @@
+		// 'GITHUB_PERSONAL_ACCESS_TOKEN=$GITHUB_PERSONAL_ACCESS_TOKEN'), let's
+		// take the value from the environment variable.
+		for index, env := range config.Env {
+			tokens := strings.SplitN(env, "=", 2)
+			if len(tokens) == 2 && len(tokens[1]) > 0 && tokens[1][0] == '$' {
+				envValue := os.Getenv(tokens[1][1:])
+				if envValue == "" {
+					logger.Errorf("Environment variable %s not found", tokens[1][1:])
+					return fmt.Errorf("MCP configuration error")
+				}
+				config.Env[index] = fmt.Sprintf("%s=%s", tokens[0], envValue)
+			}
+		}
+
+		if config.SSE != "" {
+			logger.Infof("[+] initMCPClient(%s): Using SSE transport to %s\n", name, config.SSE)
+
+			client, err := client.NewSSEMCPClient(config.SSE)
+			if err != nil {
+				logger.Errorf("Failed to create client: %v", err)
+				return fmt.Errorf("MCP configuration error")
+			}
+
+			err = client.Start(context.TODO())
+			if err != nil {
+				logger.Fatalf("Failed to start client: %v", err)
+				return fmt.Errorf("MCP configuration error")
+			}
+			config.Client = client
+		} else if config.Command != "" {
+			logger.Infof("[+] initMCPClient(%s): Using stdio transport for command %s\n", name, config.Command)
+
+			client, err := client.NewStdioMCPClient(config.Command, config.Env, config.Args...)
+			if err != nil {
+				logger.Errorf("Failed to create client: %v", err)
+				return fmt.Errorf("MCP configuration error")
+			}
+			config.Client = client
+		} else {
+			logger.Errorf("Either 'sse' or 'command' must be provided for the MCP Server configuration")
+			return fmt.Errorf("MCP configuration error")
+		}
+
+		logger.Infof("[+] Initializing client")
+		initRequest := mcp.InitializeRequest{}
+		initRequest.Params.ProtocolVersion = mcp.LATEST_PROTOCOL_VERSION
+		initRequest.Params.ClientInfo = mcp.Implementation{
+			Name:    "api-bridge-agent",
+			Version: "1.0.0",
+		}
+		initResult, err := config.Client.Initialize(context.TODO(), initRequest)
+		if err != nil {
+			logger.Errorf("Failed to initialize: %v", err)
+			return fmt.Errorf("MCP Initialization error")
+		}
+		logger.Infof(
+			"[+] Initialized with server: %s %s\n\n",
+			initResult.ServerInfo.Name,
+			initResult.ServerInfo.Version,
+		)
+
+		logger.Infof("[+] Listing available tools...")
+		toolsRequest := mcp.ListToolsRequest{}
+		tools, err := config.Client.ListTools(context.TODO(), toolsRequest)
+		if err != nil {
+			logger.Errorf("Failed to list tools: %v", err)
+			return fmt.Errorf("MCP initialization error")
+		}
+		config.Tools = tools.Tools
+		for _, tool := range tools.Tools {
+			logger.Infof("   - %s: %s\n", tool.Name, tool.Description)
+		}
+	}
+
+	return nil
+}
+
+func loadMCPPluginConfig(r *http.Request) error {
+	logger.Debugf("[+] loadMCPPluginConfig ...")
+
+	apidef := getOASDefinition(r)
+	middleware := apidef.GetTykMiddleware()
+	if middleware == nil {
+		err := fmt.Errorf("Tyk middleware definition is nil")
+		logger.Errorf("[+] initPluginFromRequest: %s", err)
```

Comment:
here and below: copy/paste error. function is loadMCPPluginConfig(), not initPluginFromRequest. Important because the two plugins are active at same time, and if an error occurred, we will not be able to know which plugin is impacted.
Better if we can choose a message that contains "MCP" to easily identify it?